srp.go

```diff
@@ -247,32 +247,46 @@ func computeMultiplier(generator, modulus *big.Int, bitLength int) (*saferith.Na
 	return new(saferith.Nat).SetBig(multiplier, bitLength), nil
 }
 
-func checkParams(bitLength int, ephemeral, generator, modulus *big.Int, modulusMinusOne *big.Int) error {
+func checkParams(bitLength int, ephemeral, generatorTwo, modulus *big.Int, modulusMinusOne *big.Int) error {
```

Comment:
Can we check that `generatorTwo == 2`, and throw on different values as unsupported ? Then I think we could leave it named `generator` .

Reply:
I'm not sure how to validate those without removing the (very minimal) benefit of having those parameters at all, namely reusing the computation and memory of those `big.Int`s. Checking `generator == 2` is super cheap and easy though. How about entirely removing `modulusMinusOne` as a parameter instead?

Reply:
The function is not exported and the library has two as the hardcoded generator, so checking generator == 2 should be fine.
Giving modulusMinusOne as a parameter was just an optimisation, because it's also used in the rest of the srp computation.